Description: escape url.toString() before passing it to QRegularExpression
 Otherwise, when the path contains a plus symbol (as in 5.14.1+dfsg),
 it does not match against self.
Author: Dmitry Shachnev <mitya57@debian.org>
Forwarded: https://codereview.qt-project.org/c/qt/qtdeclarative/+/291218
Last-Update: 2020-02-19

--- a/tests/auto/qml/qqmlapplicationengine/tst_qqmlapplicationengine.cpp
+++ b/tests/auto/qml/qqmlapplicationengine/tst_qqmlapplicationengine.cpp
@@ -302,7 +302,8 @@ void tst_qqmlapplicationengine::failureT
     auto url = testFileUrl("invalid.qml");
     qRegisterMetaType<QList<QQmlError>>();
     QTest::ignoreMessage(QtMsgType::QtWarningMsg, "QQmlApplicationEngine failed to load component");
-    QTest::ignoreMessage(QtMsgType::QtWarningMsg, QRegularExpression(url.toString() + QLatin1Char('*')));
+    QTest::ignoreMessage(QtMsgType::QtWarningMsg,
+                         QRegularExpression(QRegularExpression::escape(url.toString()) + QLatin1Char('*')));
     QQmlApplicationEngine test;
     QSignalSpy warningObserver(&test, &QQmlApplicationEngine::warnings);
     test.load(url);
